src/de/mod.rs

```diff
@@ -39,6 +39,13 @@ impl<'de> Deserializer<'de> {
     pub fn remainder(&self) -> Cow<'_, str> {
         String::from_utf8_lossy(&self.bytes.bytes())
     }
+
+    fn fix_position(&self, e: Error) -> Error {
```

Comment:
Ok, let's just make it so the input error type and the output error type are *different*. This will still require us to call it, but there will be no way to miss a fix step.

Reply:
That sounds smart, I'll give that a try